src/components/Applications/ApplicationListView.tsx

```diff
@@ -128,6 +128,7 @@ const ApplicationListView: React.FC<React.PropsWithChildren<unknown>> = () => {
                         <Link
                           {...props}
                           to={IMPORT_PATH.createPath({ workspaceName: namespace })}
+                          state={{ isApplication: true }}
```

Comment:
when user clicks on create component, we add a query param `application` to the url. Use url param to identify if it's a component flow or application flow.